### DISCUSSION TITLE
Node.js binary for RISC-V architecture.?

### DISCUSSION THREAD

**Discussion by gkdlight** [nodejs/help#4740] [Q&A] [answered]
Is there any availability of a pre-built Node.js binary specifically for the RISC-V architecture.  Currently, Node.js provides pre-built binaries for various architectures like ARM (e.g., node-v20.3.0-linux-armv7l.tar.gz) which greatly simplify the installation process.

i would like to know any  pre-built Node.js binary for the RISC-V or provide steps to build node js binary from source on linux x86.

**Comment by richardlau** [comment, marked as answer]
There are unofficial builds for RISC-V on https://unofficial-builds.nodejs.org/. https://github.com/nodejs/unofficial-builds/tree/main/recipes/riscv64 contains the scripts used to build them.

**Comment by gkdlight** [comment]
thank you richardlau,

I have reviewed the unofficial RISC-V packages on https://unofficial-builds.nodejs.org/ and noticed that they are only available starting for RISC-V architecture from Node.js version 17.7.0. However, I am currently using Node.js version 16. Is there any available package for Node.js 16 that would be compatible with the RISC-V architecture? This would greatly facilitate my work. Thank you!

**Reply by richardlau** [reply]
No, there are no RISC-V binaries for Node.js 16 AFAIK. I suspect the version of V8 in Node.js 16 doesn't support that architecture.
